overlord/devicestate/handlers.go

```diff
@@ -57,6 +57,50 @@ func (m *DeviceManager) doMarkSeeded(t *state.Task, _ *tomb.Tomb) error {
 	return nil
 }
 
+func isSameAssertsRevision(err error) bool {
+	if e, ok := err.(*asserts.RevisionError); ok {
+		if e.Used == e.Current {
+			return true
+		}
+	}
+	return false
+}
+
+func (m *DeviceManager) doSetModel(t *state.Task, _ *tomb.Tomb) error {
+	st := t.State()
+	st.Lock()
+	defer st.Unlock()
+
+	var modelass []byte
+	if err := t.Get("new-model", &modelass); err != nil {
+		return err
+	}
+
+	ass, err := asserts.Decode(modelass)
+	if err != nil {
+		return err
+	}
+
+	model, ok := ass.(*asserts.Model)
+	if !ok {
+		return fmt.Errorf("new-model is not a model assertion but: %s", ass.Type().Name)
+	}
+
+	err = assertstate.Add(st, ass)
+	if err != nil && !isSameAssertsRevision(err) {
+		return err
+	}
+
+	// TODO: look at serial/keyid/sessionmacaroonq
+	device, err := auth.Device(st)
+	if err != nil {
+		return err
+	}
+
+	// set device,model from the new model assertion
+	return setDeviceFromModelAssertion(st, device, model)
```

Comment:
this bit feels slightly premature,  for now we don't change brand/model

Reply:
I can remove it again if you feel it should not (yet) be there.

Reply:
yes, it's premature (we would need to install the new serial that we acquired very early in this case)